src/impl/realm_coordinator.cpp

```diff
@@ -462,6 +467,64 @@ void RealmCoordinator::pin_version(VersionID versionid)
     }
 }
 
+#if REALM_ENABLE_SYNC
+void RealmCoordinator::register_partial_sync_query(Realm& realm, ResultsNotifier& notifier, Query query, std::string subscription_name)
+#else 
+void RealmCoordinator::register_partial_sync_query(Realm& realm, ResultsNotifier&, Query, std::string)
+#endif
+{
+    if (!realm.is_partial())
+        throw std::logic_error("A partial sync query can only be registered in a partially synced Realm");
+
+#if REALM_ENABLE_SYNC
+    std::string object_class = query.get_table()->get_name().substr(6);
+
+    if (realm.schema().find(object_class) == realm.schema().end())
+        throw std::logic_error("A partial sync query can only be registered for a type that exists in the Realm's schema");
+
+    auto& self = Realm::Internal::get_coordinator(realm);
+    std::lock_guard<std::mutex> l(self.m_partial_sync_queue_mutex);
+    std::string key = subscription_name;
+    std::string serialized_query = query.get_description();
```

Comment:
What happens with a query that cannot be serialized?

Reply:
Currently, this method cannot crash. Core will adjust their semantics so a proper exception is thrown

Reply:
And when the exception is thrown here, how will it be handled?